com.unity.netcode.gameobjects/Tests/Runtime/MultiInstanceHelpers.cs

```diff
@@ -9,6 +9,58 @@
 
 namespace Unity.Netcode.RuntimeTests
 {
+    public interface IMultiInstanceHelperOverrides
+    {
+        bool OnCreateNewClients(int clientCount, out NetworkManager[] clients);
+
+        bool OnCreateHostServer(out NetworkManager server);
+    }
```

Comment:
huh? what's going on here?

Reply:
This provides developers with the ability to override the client and host-server creation process for the MultiInstanceHelpers class by implementing IMultiInstanceHelperOverrides and assigning the implementation to MultiInstanceHelpers.MultiInstanceHelperOverrides.  There have been many instances where this capability would be useful in the past, but in order to be able to test it does require nesting the NetworkManager which needs to be done prior to initialization.

Reply:
why it's in this PR?

Reply:
It is for testing.  It provides the ability to override how we construct the server and client NetworkManagers.
This was the solution I came up with to test a nested NetworkManager GameObject.  It also provides developers with this added functionality.

This might change as we might just throw an exception under this condition and as such would be an editor specific test as opposed to runtime (i.e. no need for using any kind of MultiInstance test).